### DISCUSSION TITLE
Will the chroot approach continue to be supported in the future?

### DISCUSSION THREAD

**Discussion by mpwyn** [ltsp/ltsp#638] [General] [answered]
We're in the process of switching over from LTSP5 to the new version. We've been using the chroot approach for years. While reading the new documentation me and my colleagues got the impression that chroot no longer plays the central role in ltsp that it used to, or at least that we alway thought it did.

So we're wondering whether the developers are maybe considering discontinuing chroot support any time in the near or far future? If so we might have to consider changing our approach. Otherwise we'll just keep on deploying chroot and learn to use the new ways of working with it.

**Comment by mpwyn** [comment]
I noticed on https://github.com/ltsp/ltsp/wiki/chroots the following:

`The primary LTSP developer recommends admins use raw VM disk images instead of using chroots as the basis for creating LTSP images because there are known issues with installing certain packages when apt and dpkg are run in a chroot.`

What sorts of issues? We've never noticed anything.

**Comment by alkisg** [comment, marked as answer]
I didn't write that sentence in the wiki! :)

Chroots are fine for sysadmins that can handle issues like:

- The XXX proprietary program installer requires Xorg; how can I run Xorg inside the chroot to click its accept button?
- dbus-daemon failed to install inside the chroot because it was trying to start a service

LTSP no longer ships tools for creating (ltsp-build-client) or managing (ltsp-chroot) chroots.
Many other tools exist that can help you with that, managing chroots isn't an LTSP-specific task.

LTSP will continue to support chroots just fine as **image sources**, i.e. `ltsp image chroot-name` will always be a supported use case; it's even the recommended way to manage [Raspberry Pi chroots](https://ltsp.org/docs/installation/raspios/).

**Reply by mpwyn** [reply]
Thanks for the clarification!
